climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,811 @@
+    Plot the average impact per exposure polygon.
+
+    Parameters
+    ----------
+    imp_geom : Impact
+        Impact instance with imp_geom set (i.e. computed from exposures with polygons)
+    centered : bool, optional
+        Center the plot. The default is False.
+    figsize : (float, float), optional
+        Figure size. The default is (9, 13).
+    **kwargs : dict
+        Keyword arguments for GeoDataFrame.plot()
+
+    Returns
+    -------
+    ax:
+        matplotlib axes instance
+
+    """
+    kwargs['figsize'] = figsize
+    if 'legend_kwds' not in kwargs:
+        kwargs['legend_kwds'] = {
+            'label': f"Impact [{imp_geom.unit}]",
+            'orientation': "horizontal"
+            }
+    if 'legend' not in kwargs:
+        kwargs['legend'] = True
+    gdf_plot = gpd.GeoDataFrame(imp_geom.geom_exp)
+    gdf_plot['impact'] = imp_geom.eai_exp
+    if centered:
+        xmin, xmax = u_coord.lon_bounds(imp_geom.coord_exp[:,1])
+        proj_plot = ccrs.PlateCarree(central_longitude=0.5 * (xmin + xmax))
+        gdf_plot = gdf_plot.to_crs(proj_plot)
+    return gdf_plot.plot(column = 'impact', **kwargs)
+
+def exp_geom_to_pnt(exp, res, to_meters, disagg_met, disagg_val):
+    """
+    Disaggregate exposures with (multi-)polygons and/or (multi-)lines
+    geometries to points.
+
+    Parameters
+    ----------
+    exp : Exposures
+        The exposure instance with exp.gdf.geometry containing lines or polygons
+    res : float
+        Resolution of the disaggregation grid / distance. Can also be a
+        tuple of [x_grid, y_grid] numpy arrays. In this case, to_meters is
+        ignored. This is only possible for Polygon-only exposures.
+    to_meters : bool
+       If True, res is interpreted as meters, and geometries are projected to
+       an equal area projection for disaggregation.  The exposures are
+       then projected back to the original projections before  impact
+       calculation. The default is False.
+    disagg_met : string
+        Disaggregation method for the `value` column of the exposure gdf.
+        if 'avg', average value over points
+        if 'fix', same value for each point
+    disagg_val: float
+        if 'None', value is taken from exp.gdf.value column, else the indicated
+        value is taken for disaggregation according to disagg_met
+
+    Returns
+    -------
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first level indicating
+        membership of the original geometries of exp,
+        second for the point disaggregation within each geometries.
+
+    """
+
+    line_mask, poly_mask = _line_poly_mask(exp.gdf)
+    gdf_geom = exp.gdf.copy()
+
+    if disagg_val is not None:
+        gdf_geom.value = disagg_val
+
+    if ((disagg_val is None) and ('value' not in gdf_geom.columns)):
+        raise ValueError('There is no value column in the exposure gdf to'+
+                         ' disaggregate from. Please set disagg_val explicitly.')
```

Comment:
Probably this check should be put in the method that is really using the parameter instead of here in the exposed method maybe? Also, I agree that it can be removed here probably.

Reply:
This is the "last" function using the `disagg_val` parameter. It's therefore in the correct place. Analogously with the function `exp_geom_to_grid`

Reply:
Still - when `disagg_val` is not None, the whole calculation is kind of done on a different `Exposures` object.
Why isn't the `Exposures` of interest not given in the first place?
Instead of
```
exp_grid = exp_geom_to_grid(exp, disagg_val=disagg_val)
```
one could always
```
exp.gdf.value = disagg_val
exp_grid = exp_geom_to_grid(exp)
```
And if the original exposures must be kept (for what reason?), the user can also take care of that.
Bottom line: the function seems to be doing too many things at a time.

Reply:
I wouldnt want to assign a value to the non-disaggregated (original) exp.gdf, as very often I have to experiment with different resolutions for disaggregation and hence different values that need to be set.
In order to avoid assigning errors (say, once i tried to disaggregate to 300m, and hence set the value of each point to 300, next time i try to disaggregate to 500m but forget to set the value to 500 ) and to avoid meaningless assignments (neither 300 nor 500 mean anything as a value on the original gdf, where my objects have arbitrary lengths / areas), i advocate for leaving it this way.